lib/kms/common.js

```diff
@@ -0,0 +1,122 @@
+import crypto from 'crypto';
+import { errors } from 'arsenal';
+
+export default class common {
+
+    static _algorithm() {
+        return 'aes-256-ctr';
+    }
+
+    /* AES-256 Key */
+    static _keySize() {
+        return 32;
+    }
+
+    /* IV is 128bit for AES-256-CTR */
+    static _IVSize() {
+        return 16;
+    }
+
+    /* block size is 128bit for AES-256-CTR */
+    static _aesBlockSize() {
+        return 16;
+    }
+
+    static createDataKey() {
+        return new Buffer(crypto.randomBytes(this._keySize()));
+    }
+
+    static _incrementIV(derivedIV, counter) {
+        const newIV = derivedIV;
+        const len = derivedIV.length;
+        let i = len - 1;
+        let ctr = counter;
+        while (ctr !== 0) {
+            const mod = (ctr + newIV[i]) % 256;
+            ctr = Math.floor((ctr + newIV[i]) / 256);
+            newIV[i] = mod;
+            i -= 1;
+            if (i < 0) {
+                i = len - 1;
+            }
+        }
+        return newIV;
+    }
+
+    /*
+     * cb(err, derivedKey, derivedIV)
+     */
+    static _deriveKey(cryptoScheme, dataKey, log, cb) {
+        if (cryptoScheme <= 1) {
+            /* we are not storing hashed human password.
+             * It's a random key, so 1 iteration and
+             * a fixed salt is enough for our usecase.
+             * don't change these without bumping
```

Comment:
instead of 'these': the salt, the iteration number or the digest algorithm (sha1 here) without bumping the scheme number saved in the object metadata along with the dataKey.